lib/commands/exam.js

```diff
@@ -239,14 +242,24 @@ module.exports = TestCommand.extend({
     const testemEvents = this.testemEvents;
     const ui = this.ui;
     const events = config.custom_browser_socket_events || {};
-    const testExecutionFileName = `test-execution-${Date.now()}.json`;
-    const browserExitHandler = function() {
+
+    const browserExitHandler = function(failed = false) {
+      if (!failed && commands.get('loadBalance')) {
+        let browserId;
+        try {
+          browserId = getBrowserId(this.launcher.settings.test_page);
+          log.info(`A browser ${browserId} exiting... [ # of modules in current module queue ${testemEvents.stateManager.getTestModuleQueue().length} ]`);
+        } catch(err) {
+          throw new Error(`A browser Id is not available. A browser is exiting.. [ # of modules in current module queue ${testemEvents.stateManager.getTestModuleQueue().length} ]`);
```

Comment:
Can we add the message here to the error to be thrown so we don't lose the actual error that's thrown?